chain/client/src/stateless_validation/chunk_endorsement_tracker.rs

```diff
@@ -112,65 +108,117 @@ impl ChunkEndorsementTracker {
         endorsement: ChunkEndorsement,
     ) -> Result<(), Error> {
         let _span = tracing::debug_span!(target: "client", "process_chunk_endorsement", chunk_hash=?chunk_header.chunk_hash()).entered();
-        self.process_chunk_endorsement_impl(endorsement, Some(chunk_header))
+
+        // Validate the endorsement before locking the mutex to improve performance.
+        if !self.epoch_manager.verify_chunk_endorsement(&chunk_header, &endorsement)? {
+            tracing::error!(target: "client", ?endorsement, "Invalid chunk endorsement.");
+            return Err(Error::InvalidChunkEndorsement);
+        }
+        self.inner.lock().unwrap().process_chunk_endorsement_impl(
```

Comment:
nit. remove _impl suffix (similarly to process_pending_endorsements of the outer calling that of inner)

Reply:
I like the `_impl` suffix because it clearly differentiates between internal functions on the `_inner` struct and public functions on the public struct.